test/vmtests/vmtests/test_min_change.py

```diff
@@ -93,39 +124,66 @@ def run_min_change_test(
     # Start VM.
     vm.start()
 
-    # Wait for VM to boot by waiting for it to request an IP address from the DHCP server.
-    vm_ip_address = vm.get_vm_ip_address(timeout=30)
-
-    # Connect to VM using SSH.
-    ssh_known_hosts_path = test_temp_dir.joinpath("known_hosts")
-    open(ssh_known_hosts_path, "w").close()
-
-    with SshClient(vm_ip_address, key_path=ssh_private_key_path, known_hosts_path=ssh_known_hosts_path) as vm_ssh:
-        vm_ssh.run("cat /proc/cmdline").check_exit_code()
-
-        os_release_path = test_temp_dir.joinpath("os-release")
-        vm_ssh.get_file(Path("/etc/os-release"), os_release_path)
-
-        with open(os_release_path, "r") as os_release_fd:
-            os_release_text = os_release_fd.read()
-
-            if input_image_azl_release == 2:
-                assert ("ID=mariner" in os_release_text)
-                assert ('VERSION_ID="2.0"' in os_release_text)
-            elif input_image_azl_release == 3:
-                assert ("ID=azurelinux" in os_release_text)
-                assert ('VERSION_ID="3.0"' in os_release_text)
-            else:
-                assert False, "Unexpected image identity in /etc/os-release"
-
-
+    # arm64 emulated runs take a very long time to boot and get to a state
+    # where we can connect to it.
+    ip_wait_time = 30
+    if platform.machine() == 'aarch64':
+        ip_wait_time = 300
+
+    # For arm64 runs, we are seeing a behavior where the first IP address that
```

Comment:
Please move this SSH connection loop into a helper function.

We will eventually expand this test suite to cover a larger number of features. And stuff like this will likely be needed by all the tests. So, it would be good to factor it out now.

Reply:
fixed